File: 1.Data structure/4.Hashmap/완주하지 못한 선수.java
import java.util.HashMap;

class Solution {
    public HashMap toHash(HashMap<String,Integer> map, String[] list){
        for(String i : list){
            if (map.containsKey(i)){
                map.put(i,map.remove(i)+1);
            }else{
                map.put(i,1);
            }
        }
        return map;
    } 
    
    public String solution(String[] participant, String[] completion) {
        HashMap<String,Integer> map = new HashMap<>();
        map = toHash(map,participant);
        map = toHash(map,completion);
        for(String key : map.keySet()){
            if (map.get(key) % 2 == 1){
                return key;
            }
        }
        return null;
    }
}

File: 1.Data structure/4.Hashmap/완주하지 못한 선수.py
def solution(participant, completion):

    def dic(todic):
        counts = {}
        for x in todic:
            if x in counts:
                counts[x] += 1
            else: counts[x] = 1
        return counts

    a = dic(participant)
    b = dic(completion)
    
    for i in a:
        try:
            if a[i] != b[i]:
                return i
        except:
            return i